src/loadsave.jl

```diff
@@ -1,37 +1,119 @@
+    path = Base.find_in_path(string(pkg)) # hacky way to determine if a Package is installed
+    path == nothing && return false
+    return isfile(path)
+end
+
+function save_import(pkg::Symbol)
+    !is_installed(pkg)      && throw(NotInstalledError(pkg, ""))
+    !isdefined(Main, pkg)   && eval(Main, Expr(:import, pkg))
+    return Main.(pkg)
+end
+
+
+for (applicable_, add_, dict_) in (
+        (:applicable_loaders, :add_loader, :sym2loader),
+        (:applicable_savers,  :add_saver,  :sym2saver))
     @eval begin
-        $appl{sym}(::Formatted{DataFormat{sym}}) = get($dct, sym, [:FileIO]) # if no loader is declared, fallback to FileIO
-        function $fchk(pkg::Symbol)
-            !isdefined(Main, pkg) && eval(Main, Expr(:import, pkg))
-            return Main.(pkg)
-        end
-        function $fadd{sym}(::Type{DataFormat{sym}}, pkg::Symbol)
-            list = get($dct, sym, Symbol[])
-            $dct[sym] = push!(list, pkg)
+        $applicable_{sym}(::@compat(Union{Type{DataFormat{sym}}, Formatted{DataFormat{sym}}})) = get($dict_, sym, [:FileIO]) # if no loader is declared, fallback to FileIO
+        function $add_{sym}(::Type{DataFormat{sym}}, pkg::Symbol)
+            list = get($dict_, sym, Symbol[])
+            $dict_[sym] = push!(list, pkg)
         end
     end
 end
 
-applicable_mime{sym}(::MIME{sym}) = get(mimedict, sym, [:nothing])
-function check_mime(pkg::Symbol)
-    pkg == :nothing && error("No MIME package available")
-    !isdefined(Main, pkg) && eval(Main, Expr(:import, pkg))
-    return pkg
-end
+applicable_mime{sym}(::MIME{sym}) = get(mimedict, sym, [(Any, :FileIO)])
 
 @doc """
 `add_mime(mime, T, :Package)`  triggers `using Package` before attempting to write object of type `T` in format `mime`.
 """ ->
 function add_mime{sym,T}(::MIME{sym}, ::Type{T}, pkg::Symbol)
-    list = get(mimedict, sym, Any[])
+    list = get(mimedict, sym, @compat(Tuple{DataType, Symbol})[])
     mimedict[sym] = push!(list, (T,pkg))
 end
 
 @doc "`add_loader(fmt, :Package)` triggers `using Package` before loading format `fmt`" -> add_loader
 @doc "`add_saver(fmt, :Package)` triggers `using Package` before saving format `fmt`" -> add_saver
+
+
+@doc """
+- `load(filename)` loads the contents of a formatted file, trying to infer
+the format from `filename` and/or magic bytes in the file.
+- `load(strm)` loads from an `IOStream` or similar object. In this case,
+the magic bytes are essential.
+- `load(File(format"PNG",filename))` specifies the format directly, and bypasses inference.
+- `load(f; options...)` passes keyword arguments on to the loader.
+""" ->
+function load(s::@compat(Union{AbstractString,IO}), args...; options...)
+    q           = query(s)
+    libraries   = applicable_loaders(q)
+    failures    = Any[]
+    for library in libraries
+        try
+            Library = save_import(library)
+            return Library.load(q, args...; options...)
```

Comment:
> FileIO gets out of the way once the appropriate package is already loaded

I'm not sure to what extend you mean this. You will always need to call `FileIO.load(::AbstractString)`, and do the query, right? If the library is already loaded, it actually doesn't get to the point to look it up in the path with this PR. It almost immediately dispatches it to the library in that case. 
There was never a direct dispatch in FileIO, was there? Even in the original design you needed to drop down to the actual IO library and do a `forced load` by knowing the correct `format`... Which you still can, right!? Okay, because of `import` vs `using`, you don't suddenly have them in your namespace. I sort of feel like this is a good thing though :D 
I'm not sure how the dictionary solution would look like without multiple dispatch...
What would you look up, when not a datatype to dispatch on? Or would you assume, that every IO library only has one function? This might be true for C-libraries but not for e.g. MeshIO which is pure Julia.

Right now it seems a little difficult to agree on anything... I'll sleep over this and see what the morning brings :)